### ISSUE TITLE
See github pull requests when changing branch

### ISSUE THREAD

**Issue by ibizaman** [shuber/vim-promiscuous#10]
_First of all, this plugin changed my life. No kidding. So thank you for making this._

I don't really like github's PR viewer so what I do is I have three aliases in my .gitconfig:

```
lolu = !sh -c 'git log --graph --pretty=format:%Cred%h%Creset\\ -%C\\(yellow\\)%d%Creset\\ %s\\ %Cgreen\\(%ad\\)\\ %C\\(bold\\ blue\\)\\<%an\\>%Creset --abbrev-commit --date=relative origin/master..HEAD' -
pullb = !sh -c 'git fetch origin pull/$0/head:pull/$0 && echo pull/$0'
pulld = !sh -c 'git branch -D pull/$0 && echo pull/$0'
```

Then I do:
1. I go to github to look at the PR and copy paste
2. I run `:!git pullb 42` and it creates a branch referencing the branch of the PR I'm interested in.
3. I run Promiscuous and switch to that branch.
4. I review it, and add comments on github.
5. With `:!git lolu` I can see the commits of the PR only.
6. I use Promiscuous to switch back to my previous branch
7. Run `:!git pulld 42` to delete the PR branch.

It would be awesome if I could make it work this way:
1. Run Promiscuous and it shows all branches + github pull request branches with author and title
2. Switch to a PR branch and Promiscuous creates on the fly.
3. I review it, and add comments on github.
4. I use Promiscuous to switch back to my previous branch which deletes that temporary branch.

That makes all the cumbersome steps automated in a pleasant way.

I really see two parts here:
- Promiscuous shows branches coming from another source.
- Promiscuous does something specific depending on the branch. Sort of like a pre-change and post-change hooks? (here it would run the `pullb` and `pulld` commands)

So this is not really a feature request but more a idea feasibility request. Do you think this would be difficult to do?

**Comment by shuber** [comment]
Interesting, I like the general concept of this and the support for "hooks". I'm going to think on this one a bit more to see if it can be done in a flexible and reusable way (e.g. support other future hooks). Thanks for the feedback, glad it's working well for you @ibizaman!

**Comment by ibizaman** [comment]
Having hooks for this is not needed thanks to https://gist.github.com/piscisaureus/3342247.

TL; DR
In `.git/config`:
```
[remote "origin"]
	...
	fetch = +refs/pull/*/head:refs/remotes/origin/pr/*
```

Then you can access all PRs with branches prefixed with `pr/`.
